dealii@develop build failure due to dependency on taskflow

https://gitlab.com/xsdk-project/spack-xsdk/-/jobs/5541203709

```
  >> 187    /data/xsdkci/VS1mJQ1K/1/xsdk-project/spack-xsdk/spack-stage/balay/s
            pack-stage-taskflow-2.7.0-2iftndqyno6k3tjhb4riltpisx6qo4ar/spack-sr
            c/3rd-party/doctest/doctest.h:4032:47: error: size of array 'altSta
            ckMem' is not an integral constant-expression
     188     4032 |         static char             altStackMem[4 * SIGSTKSZ];
     189          |                                               ^
     190    compilation terminated due to -Wfatal-errors.
```

Looks like this  is a new dependency on taskflow from dealii.

For now switching CI to `dealii~taskflow`

--- a/var/spack/repos/builtin/packages/dealii/package.py
+++ b/var/spack/repos/builtin/packages/dealii/package.py
@@ -207,7 +207,7 @@ class Dealii(CMakePackage, CudaPackage):
     depends_on("sundials@:3~pthread", when="@9.0:9.2+sundials")
     depends_on("sundials@5:5.8", when="@9.3:9.3.3+sundials")
     depends_on("sundials@5:", when="@9.3.4:+sundials")
-    depends_on("taskflow", when="@9.6:+taskflow")
+    depends_on("taskflow@3.4:", when="@9.6:+taskflow")
     depends_on("trilinos gotype=int", when="+trilinos@12.18.1:")
     # TODO: next line fixes concretization with trilinos and adol-c
     depends_on("trilinos~exodus", when="@9.0:+adol-c+trilinos")

